catalyst-gateway/bin/src/utilities/blake2b_hash.rs

```diff
@@ -0,0 +1,134 @@
+//! Types of Blake-2b Hash
+
+use blake2b_simd::Params;
+
+/// Generates a UUID string from the provided key and data using `BLAKE2b` hashing.
+///
+/// # Arguments
+/// - `key`: A string slice that is used as part of the hash function input.
+/// - `data`: A vector of strings which will be included in the `BLAKE2b` hash
+///   computation.
+///
+/// # Returns
+/// A UUID string generated from the `BLAKE2b` hash of the concatenated data with the key.
+pub(crate) fn generate_uuid_string_from_data(key: &str, data: Vec<String>) -> String {
+    // Where we will actually store the bytes we derive the UUID from.
+    let mut bytes: uuid::Bytes = uuid::Bytes::default();
+
+    // Generate a unique hash of the data.
+    let mut hasher = Params::new()
+        .hash_length(bytes.len())
+        .key(key.as_bytes())
+        .personal(b"Project Catalyst")
+        .to_state();
+
+    for datum in data {
+        hasher.update(datum.as_bytes());
+    }
+
+    // Finalize the hash and get the digest as a byte array
+    let hash = hasher.finalize();
+
+    // Create a new array containing the first 16 elements from the original array
+    bytes.copy_from_slice(hash.as_bytes());
+
+    // Convert the hash to a UUID
+    uuid::Builder::from_custom_bytes(bytes)
+        .as_uuid()
+        .as_hyphenated()
+        .to_string()
+}
+
+/// 224 Byte Blake2b Hash
+pub(crate) type Blake2b224 = [u8; 28];
```

Comment:
In all our other code we made such hash types as follows, so it will make a strong type binding. But dont have a strict opinion here.
```Rust
struct Blake2b224([u8; 28]);
```

Reply:
At this point its just a typing convenience, not a strict binding, because absolutely everything that wants to use it currently, wants the [u8; 28] anyway. so you end having `.0` or to_bytes, but then that often gives a slice, which things object to when they want the actual array itself.

If there becomes something we can do with it other than just always treat it as an array of 28 bytes, then we can revisit it.